src/lite-yt-embed.js

```diff
@@ -73,6 +77,26 @@ class LiteYTEmbed extends HTMLElement {
         document.head.append(linkEl);
     }
 
+    /**
+     * Check WebP support for the user
+     */
+    static checkWebPSupport() {
+      if (typeof LiteYTEmbed.hasWebPSupport !== 'undefined')
+        return Promise.resolve(LiteYTEmbed.hasWebPSupport);
+  
+      return new Promise(resolve => {
+        const resolveAndSaveValue = value => {
+          LiteYTEmbed.hasWebPSupport = value;
+          resolve(value);
+        }
+  
+        const img = new Image();
+        img.onload = () => resolveAndSaveValue(true);
+        img.onerror = () => resolveAndSaveValue(false);
+        img.src = 'data:image/webp;base64,UklGRh4AAABXRUJQVlA4TBEAAAAvAAAAAAfQ//73v/+BiOh/AAA=';
```

Comment:
There are better techniques on checking webp support like
```
document.createElement("canvas").toDataURL("image/webp").indexOf("data:image/webp") === 0
```

the example above will actually add new network request and will fail if not whitelisted in CSP `img-src data:`.
cc @paulirish

Reply:
Before adding the one from the PR, I tried that one you're mentioning.

Unfortunately, seems to be problematic with Safari and Firefox.

Anyway, I'm more than happy to change it if the solution based on canvas is more suitable according to @paulirish 👍

Reply:
You could simply use html5 approach

```
<picture>
    <source srcset=poster.avif type=image/avif>
    <source srcset=poster.webp type=image/webp>
    <img src=poster.jpg>
</picture>
```

with it can cover more formats, also AVIF.

Reply:
@laukstein 

```
document.createElement("canvas")
          .toDataURL("image/webp")
          .indexOf("data:image/webp") === 0
```

This is returning false on Safari while it supports `webp`. It seems that is not a good solution. :/

And still with the HTML5 approach we would need a way to preload the image before.